students/WesleyM/Lesson4/mailroom2.py

```diff
@@ -0,0 +1,55 @@
+donors = [
+  {'name': 'Alice Adams Ron', 'donation': [20]} ,
+  {'name': 'Bob Be-Lake', 'donation': [100]},
+  {'name': 'Charles Cruz', 'donation': [30, 50, 10]},
+  {'name': 'Denise Dnice', 'donation': [10, 5]},
+  {'name': 'Edward Eduardo', 'donation': [25, 20, 20, 20, 10]}
+  ]
+ 
+
+def send_ty():
+  donor = input("Enter a donor name or input 'List' for a list of donors\n>")
+  if donor.lower() == "list":
+    for names in donors:
+        print(names['name'])
+    use_input()
+  else:
+    donation = float(input(("Enter a donation amount for {} \n>").format(donor)))
+    for names in donors:
+      if donor == names['name']:
+        names['donation'].append(donation)
+        break
+    else:
+      donors.append({'name': donor, 'donation': [donation]})
+    print("Thank you {} for your donation of ${:.2f}".format(donor, donation))
+  use_input()
+  
+def letters():
+  for name in donors:
+    print('Printing Letter to {a}'.format(a = name['name']))
+    f = open('{a}.txt'.format(a = name['name']), 'w')
```

Comment:
We'll learn more about the mechanics of this later in the course, but it's good idea to use `with` when working with files because it will automatically close them for you when you exit the with block, e.g.:
```
with open(the_filename, 'w') as outfile:
    outfile.write(something)
```